test/cppsuite/test_harness/workload/workload_tracking.h

```diff
@@ -94,12 +94,69 @@ class workload_tracking : public component {
         testutil_check(_session->create(
           _session.get(), _operation_table_name.c_str(), _operation_table_config.c_str()));
         log_msg(LOG_TRACE, "Operations tracking created");
+
+        /*
+         * Open sweep cursor. This cursor will be used to clear out obsolete data from the tracking
+         * table.
+         */
+        _sweep_cursor = _session.open_scoped_cursor(_operation_table_name.c_str());
+        log_msg(LOG_TRACE, "Tracking table sweep initialized");
     }
 
     void
-    run() override final
+    do_work() override final
     {
-        /* Does not do anything. */
+        WT_DECL_RET;
+        wt_timestamp_t timestamp;
+        int64_t collection_id;
+        const char *key;
+        char *sweep_key;
+        bool globally_visible_update_found;
+
+        key = sweep_key = nullptr;
+        globally_visible_update_found = false;
+
+        /* Take a copy of the oldest so that we sweep with a consistent timestamp. */
+        wt_timestamp_t oldest_ts = _tsm.get_oldest_ts();
+
+        /* If we have a position, give it up. */
+        testutil_check(_sweep_cursor->reset(_sweep_cursor.get()));
+
+        while ((ret = _sweep_cursor->prev(_sweep_cursor.get())) == 0) {
+            testutil_check(
+              _sweep_cursor->get_key(_sweep_cursor.get(), &collection_id, &key, &timestamp));
+            /*
+             * If we're on a new key, reset the check. We want to track whether we have a globally
+             * visible update for the current key.
+             */
+            if (sweep_key == nullptr || strcmp(sweep_key, key) != 0) {
+                globally_visible_update_found = false;
+                if (sweep_key != nullptr)
+                    free(sweep_key);
+                sweep_key = static_cast<char *>(dstrdup(key));
+            }
+            if (timestamp <= oldest_ts) {
+                if (globally_visible_update_found) {
+                    log_msg(LOG_TRACE,
+                      std::string("workload tracking: Obsoleted update, key=") + sweep_key +
```

Comment:
A lot of allocation going on here. I'm wondering if we'd prefer `sprintf` or `std::stringstream` something like that.

Another thing I realized that because `log_msg` is a function rather than a macro, we pay the cost of constructing the log string even if we don't end up printing it. That's just a general issue though, not specific to this code.

Reply:
@AlexeyAnisimovAU, @luke-pearson valid points here, we should probably discuss and address those with a new ticket.

Reply:
We should have separate method for each type of log and a macros that would call those methods after the log level check.

Reply:
But we will still need to construct the string before checking the trace level ?

Reply:
Nah the idea is that you wrap the actual call to the log function in the IF check for log level effectively avoiding it.

Reply:
It depends how macros is defined I guess?
Would it construct string in this scenario? 
`#define LOG(lvl, msg) if (lvl >= _trace_level) log_msg(lvl, msg)`

Reply:
For now do an additional check:
```
if (_trace_level == LOG_TRACE)
     log_msg(blah)

```
I'll create a ticket to improve this but it will go to the next project.

Reply:
This is definitely out of the scope of this PR.

Reply:
c9e69d2